x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/template/IndexTemplateRegistry.java

```diff
@@ -101,13 +100,6 @@ public IndexTemplateRegistry(
         this.threadPool = threadPool;
         this.xContentRegistry = xContentRegistry;
         this.clusterService = clusterService;
-        if (isDataStreamsLifecycleOnlyMode(clusterService.getSettings()) == false) {
-            this.lifecyclePolicies = getLifecycleConfigs().stream()
```

Comment:
`getLifecycleConfigs` is called by the parent constructor as a result of calling `super` in the subclass constructor. That means the call to `getLifecycleConfigs` might return NULL because the subclass constructor still didn't have a chance to initialize subclass members.